autogalaxy/analysis/analysis.py

```diff
@@ -0,0 +1,483 @@
+from astropy import cosmology as cosmo
+import numpy as np
+import pickle
+import dill
+
+import autofit as af
+from autoarray import preloads as pload
+from autoarray.exc import PixelizationException, InversionException, GridException
+from autofit.exc import FitException
+from autoarray.inversion import pixelizations as pix, inversions as inv
+from autogalaxy.galaxy import galaxy as g
+from autogalaxy.plane import plane as pl
+from autogalaxy.fit import fit
+from autogalaxy.analysis import visualizer as vis
+from autogalaxy.analysis import result as res
+
+
+class Analysis(af.Analysis):
+    def __init__(self, hyper_result=None, cosmology=cosmo.Planck15):
+
+        self.hyper_result = hyper_result
+        self.cosmology = cosmology
+
+    # def save_settings(self, paths: af.Paths):
+    #     with open(f"{paths.pickle_path}/settings.pickle", "wb+") as f:
+    #         pickle.dump(self.settings, f)
+
+    def modify_before_fit(self, model, paths: af.Paths):
+        return self
+
+
+class AnalysisDataset(Analysis):
+    def __init__(
+        self,
+        dataset,
+        hyper_result=None,
+        cosmology=cosmo.Planck15,
+        settings_pixelization=pix.SettingsPixelization(),
+        settings_inversion=inv.SettingsInversion(),
+        preloads=pload.Preloads(),
+        use_result_as_hyper_dataset=False,
+    ):
+
+        super().__init__(hyper_result=hyper_result, cosmology=cosmology)
+
+        self.dataset = dataset
+
+        if self.hyper_result is not None:
+
+            self.set_hyper_dataset(result=self.hyper_result)
+
+        else:
+
+            self.hyper_galaxy_image_path_dict = None

```

Comment:
Conditionally setting attributes in the constructor is a bit weird. If they're set in the else they ought to be set in the if